### DISCUSSION TITLE
Megadesk & Companion not working. Error: "Reading from UART timed out at byte 0!"

### DISCUSSION THREAD

**Discussion by cdelog** [gcormier/megadesk#173] [Q&A]
I love the Megadesk idea, and realllllly want mine to work! :)

I purchased the megadesk and companion. Flashed the megadesk per the instructions. It seems to work and (when on USB power) I integrated it into Home Assistant with no problem.

![HA Screenshot](https://github.com/user-attachments/assets/7bab3646-4447-4904-a4b2-b061295a85b7)

I powered on the desk, and the companion is beeping repeatedly with 3 tones of increasing frequency, on loop. Each trio is accompanied by an error message in the ESPHome browser shown in the attached image. The repeating message is:

[E] [uart:015] Reading from UART timed out at byte 0!
[W] [component 237] Component <unknown> took a long time for an operation (103ms).
[W] [component 238] Components should block for at most 30ms.

Do you know what could cause this issue and how to fix it?

NOTES:
- **I have no buttons**, so no ribbon cable is connected to the companion (the used desk did not come with buttons, and I was hoping to use Megadesk through HA with no buttons). If the buttons are UART then this might be the issue??
- I had to make the connector cable. The Megadesk Companion is powering up, which seems to suggest I wired it right. I wired pins 1,2,3 -> 1,2,3 (i.e., direct pinout, NOT reverse pinout)

![ESPHome Error 1](https://github.com/user-attachments/assets/5f8c35fa-4f49-4370-8a74-72662547075a)
![ESPHome Error 2](https://github.com/user-attachments/assets/f963be87-84be-4281-9cae-f0389a4a01e1)
![ESPHome Error 3](https://github.com/user-attachments/assets/41b0ffb0-6839-43db-9497-c31bd2edfd89)

**Comment by gcormier** [comment]
So if I understand you made your own 5-pin connector between megadesk and companion?

**Comment by cdelog** [comment]
Hey! No. I'm using the included 5-pin cable between the megadesk and companion, and the included 3-pin cable between the megadesk and the desk leg. The only cable I made was the connecting cable between the two table legs, and then I'm missing the button ribbon cable.

Everything else are factory cables.

**Reply by gcormier** [reply]
Ignoring the companion, for basic operation, the megadesk does 4 beeps when it starts up. The 4th beep only occurs once it has gotten a response from the motors. If it does not, it will reboot and attempt to re-negotiate with motors. So your repeating 3 beep loop indicates to me it can't talk to the motors.

Since you have no buttons and no cable, was this desk used? Do you know if it was broken or working previously?

**Reply by cdelog** [reply]
That's helpful information. The person I got it from said it was working before they lost the button in a move, but that's not proof.

Do you know of a way to validate the integrity of the motors without the factory controls? It passes voltage through and powers the Megadesk and Companion. Any ideas to troubleshoot the connection between the Megadesk and the motors?

Might you have the electrical schematic of the desk and motors?
